File: frontend/src/api/enrollmentChange.ts
import http from './index'

export interface EnrollmentChange {
  id: number
  studentId: number
  changeType: string
  previousStatus: string
  newStatus: string
  previousClassId: number | null
  newClassId: number | null
  changeDate: string
  reason: string
  operatorId: number
  correctedRecordId: number | null
  correctionReason: string | null
  createdAt: string
  updatedAt: string
}

export interface PageResult<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

export interface CreateEnrollmentChangeDTO {
  studentId: number
  changeType: string
  previousStatus: string
  newStatus: string
  previousClassId?: number
  newClassId?: number
  changeDate: string
  reason: string
}

export interface CorrectEnrollmentChangeDTO {
  correctedRecordId: number
  correctionReason: string
  changeType: string
  previousStatus: string
  newStatus: string
  previousClassId?: number
  newClassId?: number
  reason: string
}

export function createEnrollmentChange(data: CreateEnrollmentChangeDTO) {
  return http.post<any, { data: EnrollmentChange }>('/enrollment-changes', data)
}

export function correctEnrollmentChange(data: CorrectEnrollmentChangeDTO) {
  return http.post<any, { data: EnrollmentChange }>('/enrollment-changes/correct', data)
}

export function getEnrollmentChangeById(id: number) {
  return http.get<any, { data: EnrollmentChange }>(`/enrollment-changes/${id}`)
}

export function getTimelineByStudentId(studentId: number) {
  return http.get<any, { data: EnrollmentChange[] }>(`/enrollment-changes/student/${studentId}`)
}

export function pageEnrollmentChanges(params: {
  page: number
  pageSize: number
  studentId?: number
  changeType?: string
}) {
  return http.get<any, { data: PageResult<EnrollmentChange> }>('/enrollment-changes', { params })
}

File: backend/src/main/java/com/sims/entity/EnrollmentChange.java
package com.sims.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("enrollment_change")
public class EnrollmentChange {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long studentId;
    private String changeType;
    private String previousStatus;
    private String newStatus;
    private Long previousClassId;
    private Long newClassId;
    private LocalDate changeDate;
    private String reason;
    private Long operatorId;
    private Long correctedRecordId;
    private String correctionReason;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}
